client/my-sites/checkout/checkout/bundled-domain-notice.jsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { translate } from 'i18n-calypso';
+import Gridicon from 'components/gridicon';
+import {
+	hasDomainRegistration,
+	hasPlan,
+	hasJetpackPlan,
+	isNextDomainFree,
+} from 'lib/cart-values/cart-items';
+import { getPlan, getBillingMonthsForTerm } from 'lib/plans';
+import { REGISTER_DOMAIN } from 'lib/url/support';
+import { translationExists } from 'lib/i18n-utils';
+
+function hasBiennialPlan( cart ) {
+	const plans = cart.products
+		.map( ( { product_slug } ) => getPlan( product_slug ) )
+		.filter( Boolean );
+	const plan = plans?.[ 0 ];
+
+	try {
+		return getBillingMonthsForTerm( plan?.term ) === 24;
+	} catch ( e ) {
+		return false;
+	}
+}
+
+export default function BundledDomainNotice( { cart } ) {
+	// A dotcom plan should exist.
+	if ( ! hasPlan( cart ) || hasJetpackPlan( cart ) ) {
+		return null;
+	}
+
+	// The plan should bundle a free domain
+	if ( ! isNextDomainFree( cart ) ) {
+		return null;
+	}
+
+	// Hide non-translated text for non-English users.
+	// TODO: the following lines of code should be removed once all translations are ready.
+	if (
+		! translationExists(
+			'Purchasing a one-year subscription to a WordPress.com plan gives you one year of access to your plan’s features and one year of a custom domain name.'
+		)
+	) {
+		return null;
+	}
+
+	const domainRegistrationLink = (
+		<a href={ REGISTER_DOMAIN } target="_blank" rel="noopener noreferrer" />
+	);
+
+	const oneYearCopy = translate(
+		'Purchasing a one-year subscription to a WordPress.com plan gives you one year of access to your plan’s features and one year of a custom domain name.',
+		{
+			context: 'checkout',
+		}
+	);
+	const twoYearCopy = translate(
+		'Purchasing a two-year subscription to a WordPress.com plan gives you two years of access to your plan’s features and one year of a custom domain name.',
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?